### PR TITLE
Fix#9251: DB2 connection config and ingestion update

### LINKED ISSUE #9251: DB2 Metadatata Ingestion Error
**Affected module**
Ingestion Framework

**Describe the bug**
Hi, when connecting to DB2 (zOS) I get this error when executing "metadata ingest" command. I'm providing the user, password, hostport and databaseschema name in the YAML file, but for any reason the database name is mapped to "NONE                   ".  The return message says connector is reaching the remote db server but................. With a windows jdbc db client and same connection parameters I get the db schema.

```
[2022-12-01 18:37:35] DEBUG    {Metadata:ingest:41} - Traceback (most recent call last):
  File "/opt/informatica/openmetadata/env/lib64/python3.8/site-packages/ibm_db_dbi.py", line 629, in connect
    conn = ibm_db.connect(dsn, '', '', conn_options)
Exception: [IBM][CLI Driver] SQL30061N  The database alias or database name "NONE              " was not found at the remote node.  SQLSTATE=08004 SQLCODE=-30061
```

**To Reproduce**
image

**Expected behavior**

**Version:**
 - OS: 3.10.0-1160.80.1.el7.x86_64
 - Python version: Python 3.8.14
 - OpenMetadata version: 0.12.1
 - OpenMetadata Ingestion package version: openmetadata-ingestion[db2]==0.12.1.3

**Additional context**

### PR DESCRIPTION
**Pull request by nahuelverdugo**
Fixes #9251 and #7109

### Describe your changes :
DB2 connector supports multiple databases so the configuration must use the `database` instead of the `databaseSchema` to create the connection. In addition, I have updated the dependencies of `db2` connector which fixes the issue of ingesting primary and foreign keys from the tables.

image

### Type of change :
- [x] Bug fix

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/db2Connection.json [hunk 1]
```diff
@@ -48,9 +48,9 @@
       "description": "Host and port of the DB2 service.",
       "type": "string"
     },
-    "databaseSchema": {
-      "title": "databaseSchema",
-      "description": "databaseSchema of the data source. This is optional parameter, if you would like to restrict the metadata reading to a single databaseSchema. When left blank, OpenMetadata Ingestion attempts to scan all the databaseSchema.",
+    "database": {
+      "title": "database",
+      "description": "Database of the data source.",
```
Comment (ayush-shah) [inline openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/db2Connection.json:53]: any particular reason we changed the databaseSchema to database?

Comment (ayush-shah) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/db2Connection.json:53]: I think both will work similarly,
based on the code in source_connections

         if hasattr(connection, "database"):
                url += f"/{connection.database}" if connection.database else ""
        
            elif hasattr(connection, "databaseSchema"):
                url += f"/{connection.databaseSchema}" if connection.databaseSchema else ""

Comment (nahuelverdugo) [reply openmetadata-spec/src/main/resources/json/schema/entity/services/connections/database/db2Connection.json:53]: They will work similarly to create the connection but not to perform the ingestion. If we leave "databaseSchema", the database name will be `default,` and we want to start ingestion from the database name, not from a schema.

File: ingestion/src/metadata/ingestion/source/database/common_db_source.py [hunk 2]
```diff
@@ -175,7 +175,13 @@ def get_table_description(
...
             )
         else:
-            description = table_info["text"]
+            if hasattr(table_info, "text"):
+                description = table_info["text"]
+                # DB2 connector does not return a str type
+                if type(table_info["text"]) is list:
+                    description = description[0]
+                else:
+                    description = description
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/database/common_db_source.py:184]:
```suggestion
            if hasattr(table_info, "text"):
                description = table_info["text"]
                # DB2 connector does not return a str type
                if type(description) is list:
                    description = description[0]
```

File: bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql [hunk 3]
```diff
@@ -36,3 +36,11 @@ CREATE TABLE IF NOT EXISTS alert_action_def (
     PRIMARY KEY (id),
     UNIQUE (name)
 );
+
```
Comment (nahuelverdugo) [inline bootstrap/sql/com.mysql.cj.jdbc.Driver/v007__create_db_connection_info.sql:39]:
```suggestion
```

**Review by ayush-shah — APPROVED** [review]

**Review by ayush-shah — DISMISSED** [review]

**Review by harshach — DISMISSED** [review]

**Review by pmbrull — APPROVED** [review]
Thanks for the fixes and cleaning the schemas. LGTM

**Pull request merged**